scripts/update_config.js

```diff
@@ -19,77 +19,82 @@ const program = require('commander');
+	process.exit(1);
+}
 
-if (program.old) {
-	oldConfig = JSON.parse(fs.readFileSync(program.old, 'utf8'));
+console.info('Starting configuration migration...');
+const oldConfig = JSON.parse(fs.readFileSync(oldConfigPath, 'utf8'));
+const newConfig = JSON.parse(fs.readFileSync(newConfigPath, 'utf8'));
 
-	// Values to keep from new config file
-	delete oldConfig.version;
-	delete oldConfig.minVersion;
+// Values to keep from new config file
+delete oldConfig.version;
+delete oldConfig.minVersion;
 
-	// Rename old port to new wsPort
-	oldConfig.httpPort = oldConfig.port;
-	oldConfig.wsPort = oldConfig.httpPort + 1;
-	delete oldConfig.port;
+// Rename old port to new wsPort
+oldConfig.httpPort = oldConfig.port;
+oldConfig.wsPort = oldConfig.httpPort + 1;
+delete oldConfig.port;
 
-	oldConfig.db.max = oldConfig.db.poolSize;
-	delete oldConfig.db.poolSize;
+oldConfig.db.max = oldConfig.db.poolSize;
+delete oldConfig.db.poolSize;
 
-	delete oldConfig.peers.options.limits;
+delete oldConfig.peers.options.limits;
 
-	oldConfig.transactions.maxTransactionsPerQueue =
-		oldConfig.transactions.maxTxsPerQueue;
-	delete oldConfig.transactions.maxTxsPerQueue;
+oldConfig.transactions.maxTransactionsPerQueue =
+	oldConfig.transactions.maxTxsPerQueue;
+delete oldConfig.transactions.maxTxsPerQueue;
 
-	delete oldConfig.loading.verifyOnLoading;
-	delete oldConfig.dapp;
+delete oldConfig.loading.verifyOnLoading;
+delete oldConfig.dapp;
 
-	// Peers migration
-	oldConfig.peers.list.map(p => {
-		p.wsPort = p.port;
-		delete p.port;
-	});
+// Peers migration
+oldConfig.peers.list.map(p => {
+	p.wsPort = p.port;
+	delete p.port;
+});
 
-	if (oldConfig.forging.secret && oldConfig.forging.secret.length) {
-		if (!program.password.trim()) {
-			const rl = readline.createInterface({
-				input: process.stdin,
-				output: process.stdout,
-			});
-			rl.question(
-				'We found some secrets in your config, if you want to migrate, please type in your password (enter to skip): ',
-				password => {
-					rl.close();
-					migrateSecrets(password);
-					copyTheConfigFile();
-				}
-			);
-			// To Patch the password support
-			rl.stdoutMuted = true;
-			rl._writeToOutput = function _writeToOutput(stringToWrite) {
-				if (rl.stdoutMuted) rl.output.write('*');
-				else rl.output.write(stringToWrite);
-			};
-		} else {
-			migrateSecrets(program.password);
-			copyTheConfigFile();
-		}
+if (oldConfig.forging.secret && oldConfig.forging.secret.length) {
+	if (!program.password.trim()) {
+		const rl = readline.createInterface({
+			input: process.stdin,
+			output: process.stdout,
+		});
+		rl.question(
+			'We found some secrets in your config, if you want to migrate, please type in your password (enter to skip): ',
+			password => {
+				rl.close();
+				migrateSecrets(password);
+				copyTheConfigFile();
+			}
+		);
+		// To Patch the password support
+		rl.stdoutMuted = true;
+		rl._writeToOutput = function _writeToOutput(stringToWrite) {
```

Comment:
This doesn't seem like a best practice.

Reply:
Yes already commented there, its a patch, because this functionality is not supported in current node version.